docs/reference/search/search-fields.asciidoc

```diff
@@ -4,33 +4,203 @@
+loading fields:
 
 * Use the <<docvalue-fields, `docvalue_fields`>>
 parameter to get values for selected fields. This can be a good
 choice when returning a fairly small number of fields that support doc values,
 such as keywords and dates.
-* Use the <<request-body-search-stored-fields, `stored_fields`>> parameter to get the values for specific  stored fields. (Fields that use the <<mapping-store,`store`>> mapping option.)
+* Use the <<request-body-search-stored-fields, `stored_fields`>> parameter to
+get the values for specific stored fields (fields that use the
+<<mapping-store,`store`>> mapping option).
 
-You can find more detailed information on each of these methods in the 
+You can find more detailed information on each of these methods in the
 following sections:
 
-* <<source-filtering>>
+* <<search-fields-param>>
 * <<docvalue-fields>>
 * <<stored-fields>>
+* <<source-filtering>>
+
+[discrete]
+[[search-fields-param]]
+=== Fields
+
+The `fields` parameter allows for retrieving a list of document fields in
+the search response. It consults both the document `_source` and the index
+mappings to return each value in a standardized way that matches its mapping
+type. By default, date fields are formatted according to the
+<<mapping-date-format,date format>> parameter in their mappings.
+
+.*Example*
+[%collapsible]
+====
+The following search request uses the `fields` parameter to retrieve values
+for the `clientip` field, all fields starting with `location.`, and the
+`timestamp` field:
+
+[source,console]
+----
+POST logs-*/_search
+{
+  "query": {
+    "match_all": {}
+  },
+  "fields": [
+    "clientip",
+    "location.*", <1>
+    {
+      "field": "timestamp",
+      "format": "epoch_millis" <2>
+    }
+  ],
+  "_source": false
+}
+----
+
+<1> Both full field names and wildcard patterns are accepted.
+<2> Using object notation, you can pass a `format` parameter to apply a custom
+    format for the field's values. This is currently supported for
+    <<date,`date` fields>> and <<date_nanos, `date_nanos` fields>>, which
+    accept a <<mapping-date-format,date format>>.
+
+The values are returned as a flat list in the `fields` section in each hit:
+
+[source,console-result]
+----
+{
+  "took" : 2,
+  "timed_out" : false,
+  "_shards" : {
+    "total" : 1,
+    "successful" : 1,
+    "skipped" : 0,
+    "failed" : 0
+  },
+  "hits" : {
+    "total" : {
+      "value" : 1,
+      "relation" : "eq"
+    },
+    "max_score" : 1.0,
+    "hits" : [
+      {
+        "_index" : "http-logs",
+        "_id" : "1",
+        "_score" : 1.0,
+        "fields" : {
+          "clientip" : [
+            "192.0.2.0"
+          ],
+          "location.city" : [
+            "Toronto"
+          ],
+          "location.country" : [
+            "Canada"
+          ],
+          "timestamp" : [
+            "1593274413000"
+          ]
+        }
+      }
+    ]
+  }
+}
+----
+// TESTRESPONSE[skip:no test set-up]
```

Comment:
I think it'd be a little nicer to `//TEST[setup:twitter]` or something like that so we actually get a real example and the docs tests make sure the output lines up.

Reply:
That makes sense, I'll switch to the twitter test set-up.